### PR TITLE
Raincatch 1172

### PR DESCRIPTION
**Pull request by jstaffor**

### CODE REVIEW

File: docs/workforce-management-framework/topics/con-passportjs-module.adoc [hunk 1]
```diff
@@ -0,0 +1,17 @@
+[id='con-passportauth-module-{chapter}']
+= Introducing the {WFM-RC-NameShort} Passport.js Module
+
+*Passport.js* is a lightweight authentication middleware for Node.js that is extremely flexible and modular.
+*Passport.js* offers a comprehensive set of link:http://passportjs.org/docs/configure[strategies] that support authentication using a username and password, Facebook, Twitter, and more(?).
```
Comment (jstaffor) [inline docs/workforce-management-framework/topics/con-passportjs-module.adoc]: "that supports authentication using a username and password, Facebook, Twitter, and more(?)."

I don't really understand this. It " supports authentication using a username and password". "Facebook, Twitter, and more(?)." - I don't get.

File: docs/workforce-management-framework/topics/con-passportjs-module.adoc [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+*Passport.js*:
+
+* Can be bundled into an existing Node.js server
+* Is available by default when running our reference server application(?)
```
Comment (jstaffor) [inline docs/workforce-management-framework/topics/con-passportjs-module.adoc]: What is the "our reference server application(?)" - it's not mentioned anywhere else

File: docs/workforce-management-framework/topics/pro-working-with-passportjs.adoc [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+= Introduction
+
+This section includes:
+
+. xref:pro-configuring-passportjs-{chapter}[Configuring Passport.js]
+. xref:pro-implementing-the-userrepository-{chapter}[Implementing the UserRepository]
+. xref:pro-modifying-a-strategy-{chapter}[Modifying a Strategy]
+
+= Prerequisites
+(?)
```
Comment (jstaffor) [inline docs/workforce-management-framework/topics/pro-working-with-passportjs.adoc]: What are the they, if applicable?

File: docs/workforce-management-framework/topics/pro-working-with-passportjs.adoc [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+= Procedure
+
+[id='pro-configuring-passportjs-{chapter}']
```
Comment (jstaffor) [inline docs/workforce-management-framework/topics/pro-working-with-passportjs.adoc]: Can we turn the content in the 3 headings into something that resembles the delivery of actions in "docs/workforce-management-framework/topics/pro-running-the-demo-app.adoc"?

Note: this adoc is one that I didn't get to review in full as time is something I'm short on atm hence the "90% docs approved" statement during our call

**Review by jstaffor — COMMENTED** [review]
A number of question I have:

**Comment by jstaffor** [comment]
@austincunningham - I've attempt to sync docs with updates after tech-ack label. Can you take a look now? I'll have to complete a docs review again however I'll hold off until this PR is in a state where we feel it's synced.

File: docs/workforce-management-framework/topics/con-passportjs-module.adoc (hunk 2 above)
Comment (austincunningham) [reply docs/workforce-management-framework/topics/con-passportjs-module.adoc]: its a reference to the demo app. Demo app will not exist going forward as a product so we may have to figure out what to call it.

File: docs/workforce-management-framework/topics/pro-working-with-passportjs.adoc (hunk 3 above)
Comment (austincunningham) [reply docs/workforce-management-framework/topics/pro-working-with-passportjs.adoc]: same as the getting started prerequisites so shouldn't apply them.

File: docs/workforce-management-framework/topics/pro-working-with-passportjs.adoc (hunk 4 above)
Comment (austincunningham) [reply docs/workforce-management-framework/topics/pro-working-with-passportjs.adoc]: not sure what you mean , could hammer it out on a call.

**Review by austincunningham — COMMENTED** [review]
all in all looks good but I would move the getting started guide above this content as you can't use it with out running getting started first.

**Comment by jstaffor** [comment]
Final comment to address:

**1) 5.1.4. Passport.js Authentication**
 **- Text:**
"Passport.js uses authenticate middleware for accepting user credentials and creates a session entry that holds user[authenticate] middleware for accepting user credentials and creates session entry that holds user. To access routes, a user must logon using their credentials. "
 **- Comment:** This text doesn't make a lot of sense to me - defo needs to be refactored. Can you help me with this?


**Original:**
Passports using
link:../../../raincatcher-docs/{WFM-RC-Api-Version}{WFM-RC-Api-Endpoint-Security}#authenticate[authenticate]
middleware for accepting user credentials and creates session entry that holds user.
To access routes user need to login using their credentials. Cookie containing user session key will be used to authenticate and authorize user requesting server endpoint.

**Maybe change to**
Passport is using link:[authenticate] middleware for accepting user credentials and creates session entry for a user. To access Express routes a user needs to login using their credentials. Cookie containing user session key will be used to authenticate and authorize user requesting server endpoint.

**Comment by jstaffor** [comment]
I've updated the text based on feedback above and I'm happy for it to be pushed to upstream repo. Great work on this @austincunningham!

**Comment by jstaffor** [comment]
Latest push includes a host of changes based the fact that PassportAuth is the module being used by RC.
I'll test the doc in full for broken links, missing full stops, etc after you are happy with the content.

**Comment by jstaffor** [comment]
Hi @austincunningham,
- what is the current status of this PR?
- what is your plan re this PR?

**Review by austincunningham — COMMENTED** [review]
closing this PR as it is looking to merge with austincunningham/raincatcher-docs:master instead of  upstream master/ have moved the contents to https://github.com/feedhenry-raincatcher/raincatcher-docs/pull/43 so it can be reviewed by the team and see if we can get tech approval for this. If you can mark the new PR as Docs Approved . Thanks

**Pull request closed without merging**